### ISSUE TITLE
Tags selector field: the placeholder shouldn't be used as replacement for a visible label

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#10125]
For better accessibility and usability, placeholders shouldn't be used as replacement for visible labels. Some designs tend to use placeholders to "save space" or mainly for aesthetic purposes, but there are many accessibility and usability considerations to take into account. There's also a core Trac ticket about this issue, with a pretty good amount of documentation and literature linked, see https://core.trac.wordpress.org/ticket/40331. I'd strongly recommend to go through all those resources and articles to get a good grasp of what the nature of the issue is.

Worth also reminding using placeholders as replacement for labels goes against the HTML specification, where it's clearly stated placeholders should be used only for a short hint about the expected format or for a sample value. See https://www.w3.org/TR/html52/sec-forms.html#the-placeholder-attribute

For better accessibility, a visible `<label>` element is always preferable. In this specific case, there are also usability and design consistency considerations.

Most of the input fields in the sidebar have visible labels. The tags field is the only one that doesn't have a visible label and uses a placeholder text instead. Apart from accessibility concerns, I'd like to point out two main issues:

- when the field is filled with some tokens, the placeholder disappears: there's no visual indication of what this field is about, except the title of the wrapping panel 
- this field, as all the other components, is meant to be re-usable in different contexts and there's no guarantee a wrapping container will give some visual indication of what the field is about: it's very likely this field will be visually unlabelled

<img width="600" alt="screen shot 2018-09-23 at 15 18 01" src="https://user-images.githubusercontent.com/1682452/45928690-08251a80-bf48-11e8-9757-30fe231ff379.png">

A visible `<label>` element would solve all these issues and would benefit everyone, making this component actually reusable and solving all the accessibility / usability / consistency issues:

<img width="600" alt="screen shot 2018-09-23 at 12 14 49" src="https://user-images.githubusercontent.com/1682452/45928702-386cb900-bf48-11e8-99b9-68f9718ce499.png">

**Comment by aaronjorbin** [comment]
I wonder if it might make more sense to open tickets for each form field that we need to update as we may need to design each one a bit differently.
